interview-questions.md

```diff
@@ -9,41 +9,58 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. What is object-oriented programming? How is it different than functional programming?
 
 Your answer:
+Object-oriented programming is when everything we interact with is an object. They are based off classes. functional programming is making a machine that takes an input and changes it according to the code in the function and outputs something made out of the input given. 
 
-Researched answer:
+Researched answer: 
+Object-oriented programming or OOP is programming that focuses on the use of classes and objects. Both the data and logic/functionality are grouped together. OOP consists of objects, classes, methods, and attributes. Functional programming is coding with inputs and outputs with a function to pass the input to. This way of programming relies on arguments. A neat thing I found is that this process of coding is called "eliminating side effects in your code."
 
 2. What is the difference between a Float and an Integer in Ruby?
 
-Your answer:
+Your answer: 
+An interger is a digit, a number. A whole number like 1, 2, 3, 4... A float looks like a decimal (ex: 1.2) but it is its own element. It has less to do with being a numeric value and more to do with a space or a point. Like float points. 
 
-Researched answer:
+Researched answer: 
+Looking deeper, an interger is in fact a number that can be represented as a whole number, negative number, or a 0. A floats definition concerning Ruby is a number written with decimals (ex: 3.14). When you need to be more precise you would use a float. 
 
 3. Ruby has an implicit return. What does that mean?
 
-Your answer:
+Your answer: 
+In my own words an impicit return is that a return is impied thus you do not need to declare or write return to get the return functionality. My car has auto headlights, I don't need to tell it to turn them on when its dark. 
 
 Researched answer:
+An implicit return is in a method and it is the final statement. We do not need the return keyword. The result will be the same. Returns the last excecuted instruction.
 
 4. What is a block in Ruby?
 
-Your answer:
+Your answer: Blocks are like functions, and they start with do and end with end. You can pass in arguements and it will output what you asked it to. 
 
-Researched answer:
+Researched answer: A block is an anonymous function. It is encapsulated in do and end. These anonymous functions can be passed into methods. You can also save it into a variable and use it later on. 
 
-1. How do you extract a value in a Ruby hash?
+5. How do you extract a value in a Ruby hash?
 
 Your answer:
+First we need to know what a hash is. It is Ruby's version of an object. These objects have key:value pairs inside them. We would extract the value by calling on the hash and the key that holds the value. If the hash is coffee_types and the coffee you want is a pumpkin spice latte, the key would be something like specialty so you would call upon specialty 
```

Comment:
!Important! Ruby has objects, Hash is kind of ruby object. But otherwise good work1